openstf_dbc/services/prediction_job.py

```diff
@@ -4,13 +4,46 @@
 
 import json
 from typing import List, Optional, Union
+from datetime import datetime
+from pydantic import BaseModel, ValidationError
 
 from openstf_dbc.data.featuresets import FEATURESET_NAMES, FEATURESETS
 from openstf_dbc.data_interface import _DataInterface
 from openstf_dbc.log import logging
 from openstf_dbc.services.systems import Systems
 
 
+# With suggestion on how to split this
+class PredictionJobDataClass(BaseModel):
+    id: Union[int, str]  # both
+    model_type_group: str  # model_specs
+    model: str  # model_specs
+    forecast_type: str  # prediction_job
+    horizon_minutes: int  # prediction_job
+    resolution_minutes: int  # prediction_job
+    lat: float  # prediction_job
+    lon: float  # prediction_job
+    train_components: bool  # prediction_job
+    name: str  # prediction_job
+    created: datetime  # prediction_job
+    sid: Optional[str]  # prediction_job
+    hyper_params: Optional[dict]  # model_specs
+    feature_names: Optional[list]  # model_specs
+    description: Optional[str]  # prediction_job
+    quantiles: Optional[List[float]]  # model_specs
+
+    def __getitem__(self, item):
+        """Allows us to use subscription to get the items from the object"""
+        return getattr(self, item)
+
+    def __setitem__(self, key: str, value: any):
+        """Allows us to use subscription to set the items in the object"""
+        if hasattr(self, key):
+            self.__dict__[key] = value
+        else:
+            raise AttributeError(f"{key} not an attribute of prediction job")
+
+
 class PredictionJob:
```

Comment:
Perhaps good idea to rename the 'PredictionJob' class to 'PredictionJobRetriever' or something, so someone isn't confused and thinks this is the data class.